cluster/gce/util.sh

```diff
@@ -154,11 +154,23 @@ function copy-to-staging() {
   local -r gs_url=$2
   local -r tar=$3
   local -r hash=$4
+  local -r basename_tar=$(basename ${tar})
+
+  #check whether this tar alread exists and has the same hash
+  #if it matches, then don't bother uploading it again
+  local -r remote_tar_hash=$(gsutil hash -h -m ${staging_path}/${basename_tar} 2>/dev/null | grep "Hash (md5):" | awk -F ':' '{print $2}')
```

Comment:
How does this work against a remote tar?  I am not familiar with gsutil but the description is ` The hash command calculates hashes on a local file`.  Is this working on a remote file or a local file?